src/main/scala/nl/knaw/dans/easy/multideposit/package.scala

```diff
@@ -96,201 +98,216 @@ package object multideposit {
   }
 
   implicit class TryExceptionHandling[T](val t: Try[T]) extends AnyVal {
-    /** Terminating operator for `Try` that converts the `Failure` case in a value.
-      *
-      * @param handle converts `Throwable` to a value of type `T`
-      * @return either the value inside `Try` (on success) or the result of `handle` (on failure)
-      */
+    /**
+     * Terminating operator for `Try` that converts the `Failure` case in a value.
+     *
+     * @param handle converts `Throwable` to a value of type `T`
+     * @return either the value inside `Try` (on success) or the result of `handle` (on failure)
+     */
     def onError[S >: T](handle: Throwable => S): S = {
       t match {
         case Success(value) => value
         case Failure(throwable) => handle(throwable)
       }
     }
-  }
 
-  implicit class OptionExtensions[T](val option: Option[T]) extends AnyVal {
-    /**
-      * Converts an `Option` to an `Observable`. `Some` is mapped to an `Observable` with 1 element,
-      * `None` is mapped to the empty `Observable`.
-      *
-      * @return an `Observable` with either zero or one elements
-      */
-    def toObservable = option.map(Observable.just(_)).getOrElse(Observable.empty)
+    def ifSuccess(f: T => Unit): Try[T] = {
+      t match {
+        case success@Success(x) => Try {
+          f(x)
+          return success
+        }
+        case e => e
+      }
+    }
+
+    def ifFailure(f: PartialFunction[Throwable, Unit]): Try[T] = {
+      t match {
+        case failure@Failure(e) if f.isDefinedAt(e) => Try {
+          f(e)
+          return failure
+        }
+        case x => x
+      }
+    }
```

Comment:
These are for side-effects specifically. I'll move these to the dans-scala-lib some day.